cloud/openstack/os_router_facts.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_router_facts
+short_description: Retrieve facts about one or more OpenStack routers.
+version_added: "2.2"
+author: "Greg Hellings (@greg-hellings)"
+description:
+    - Retrieve facts about one or more routers from OpenStack.
+requirements:
+    - "python >= 2.6"
+    - "shade"
+options:
+   router:
+     description:
+        - Name or ID of the Router
+     required: false
+   filters:
+     description:
+        - A dictionary of meta data to use for further filtering.  Elements of
+          this dictionary may be additional dictionaries.
+     required: false
+extends_documentation_fragment: openstack
+'''
+
+EXAMPLES = '''
+# Gather facts about previously created routers
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+- debug: var=openstack_routers
+
+# Gather facts about a previously created router by name
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    name:  router1
+- debug: var=openstack_routers
+
+# Gather facts about a previously created router with filter (note: name and
+  filters parameters are Not mutually exclusive)
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    filters:
+      tenant_id: 55e2ce24b2a245b09f181bf025724cbe
+      subnets:
+        - 057d4bdf-6d4d-4728-bb0f-5ac45a6f7400
+        - 443d4dc0-91d4-4998-b21c-357d10433483
```

Comment:
This is not a valid example. Looks like you just copied/pasted from os_networks_facts.py.